### PR TITLE
feat: generate summary for an issue

### PR DESCRIPTION
**Pull request by diivi**
## Description
This supports the development of https://github.com/open-sauced/insights/pull/1496 by adding the capability to summaries issues.

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
https://github.com/open-sauced/insights/pull/1496

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

File: src/issues/issue-summary.service.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      `Generate an apt issue summary for an issue taken from github issues with the specifications mentioned below`,
+      `The summary should be written in past tense and ${tone} tone`,
+      `Summary language: ${language}`,
+      `Summary must be a maximum of ${maxLength} characters.`,
+      "Exclude anything unnecessary such as translation. Your entire response will be used to summarize and analyze the issue",
+    ].join("\n");
+  }
+
+  async generateIssueSummary(options: CreateIssueSummaryDto) {
+    const content = `Issue Title: ${options.issueTitle}\nIssue Description: ${options.issueDescription}\n`;
```
Comment (bdougie) [inline src/issues/issue-summary.service.ts:21]: We should include comments as part of the embeddings.

**Review by bdougie — CHANGES_REQUESTED** [review]
Did a first pass on reviewing. 

Two items that I think would take this from good to great.

**Comment by brandonroberts** [comment]
@diivi looks like a manual change is needed https://github.com/open-sauced/api/actions/runs/5799367256/job/15719209435?pr=245

File: src/user/services/user.service.ts [hunk 2]
```diff
@@ -164,7 +164,7 @@ export class UserService {
     try {
       const user = await this.findOneById(id, true);
 
-      if (user && !user.is_open_sauced_member) {
+      if (!user.is_open_sauced_member) {
```
Comment (diivi) [inline src/user/services/user.service.ts:167]: This wasn't a `npm run format` error btw, but I don't know how the upstream had this even before I created this PR (I didn't touch this file).

Comment (brandonroberts) [reply src/user/services/user.service.ts:167]: It was already displaying this warning. Not a result of your pr

**Review by brandonroberts — APPROVED** [review]

**Review by bdougie — APPROVED** [review]

**Pull request merged**